src/apps/lwaftr/nh_fwd.lua

```diff
@@ -89,15 +87,9 @@ end
 local function copy_ether(dst, src)
    ffi.copy(dst, src, 6)
 end
-local function copy_ipv4(dst, src)
-   ffi.copy(dst, src, 4)
-end
-local function copy_ipv6(dst, src)
-   ffi.copy(dst, src, 16)
-end
-local function get_ipv4_header_length(ptr)
-   local ver_and_ihl = ptr[0]
-   return lshift(band(ver_and_ihl, 0xf), 2)
+
+local function random_port ()
+   return math.random(65535)
```

Comment:
Yes, it is possible to use the privileged ports. When the "a bits" (https://tools.ietf.org/html/rfc7597#section-5.1) value is equals to zero, that means all the possible values in the port range are used, in other words, there are not reserved ports for a port range.

```
                             0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5
                             +-----------+-----------+-------+
               Ports in      |     A     |    PSID   |   j   |
            the CE port set  |    > 0    |           |       |
                             +-----------+-----------+-------+
                             |  a bits   |  k bits   |m bits |

            Figure 2: Structure of a Port-Restricted Port Field
```

In a lwAFTR configuration where a PSID map is equals to:

```
psid-map {
   addr 178.79.150.2;
   psid-length 10;
   shift 6;
  }
```

A softwire which psid=0 uses the port range 0-1023:

```
softwire {
   ipv4 178.79.150.2;
   psid 0;
   b4-ipv6 127:11:12:13:14:15:16:128;
}
```

If the `random_port` doesn't emit values in the range of 0-1023, that means there might be softwires which will never be tested. I prefer not to make any assumption on the how the lwAFTR is configured and emit values in the whole port range.